ebean-core/src/main/java/io/ebeaninternal/server/deploy/BeanPropertyJsonMapper.java

```diff
@@ -25,11 +24,9 @@ public BeanPropertyJsonMapper(BeanDescriptor<?> desc, DeployBeanProperty deployP
   boolean isDirtyValue(Object value, EntityBeanIntercept ebi) {
     // dirty detection based on md5 hash of json content
     final String json = scalarType.jsonMapper(value);
-    final String newHash = Md5.hash(json);
-    final String oldHash = ebi.mutableHash(propertyIndex);
-    if (!Objects.equals(newHash, oldHash)) {
+    final MutableHash oldHash = ebi.mutableHash(propertyIndex);
+    if (oldHash == null || !oldHash.isEqualToJson(json)) {
       ebi.mutableContent(propertyIndex, json); // so we only convert to json once
-      ebi.mutableHash(propertyIndex, newHash); // for dirty detection next time
```

Comment:
moved to BindablePropertyJsonUpdate, so that it is done after the persist-controllers are invoked.

Reply:
This means that the hash is computed twice which isn't ideal.